cpp/src/arrow/compute/kernels/scalar_temporal_unary.cc

```diff
@@ -1503,14 +1503,13 @@ struct ISOCalendar {
     std::unique_ptr<ArrayBuilder> array_builder;
     RETURN_NOT_OK(MakeBuilder(ctx->memory_pool(), IsoCalendarType(), &array_builder));
     StructBuilder* struct_builder = checked_cast<StructBuilder*>(array_builder.get());
-    RETURN_NOT_OK(struct_builder->Reserve(in.length));
```

Comment:
Should this be kept? (eg `struct_builder->AppendNull()` is used, which might also set the top-level bitmap?)

Reply:
I don't know how the top-level bitmap behavior here would be. Reverting the change.